src/common/os.cc

```diff
@@ -19,7 +19,7 @@ using namespace std;
 
 namespace Pistache {
 
-int hardware_concurrency() {
+uint hardware_concurrency() {
```

Comment:
https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency

std::thread::hardware_concurrency() returns an int.  I agree that it _should be_ an unsigned int, but the c++11 standard has it signed.

In fact, consider removing the entire body of this method and just "return std::thread::hardware_concurrency();".

Reply:
Maybe I'm missing something, but in the link it's specified that `std::thread::hardware_concurrency()` returns an unsigned int?

I'll change the implementation to use `std::thread::hardware_concurrency()`.